cpp/src/join/hash_join.cu

```diff
@@ -41,6 +41,68 @@ std::pair<std::unique_ptr<table>, std::unique_ptr<table>> get_empty_joined_table
   return std::make_pair(std::move(empty_probe), std::move(empty_build));
 }
 
+/**
+ * @brief Performs bitwise AND of the bitmasks of columns of `build`. Returns
+ * a pair of row bitmasks and count of valid rows.
+ *
+ * @param build Table of columns used to build join hash.
+ * @param nulls_equal Flag to denote nulls are equal or not.
+ * @param stream CUDA stream used for device memory operations and kernel launches.
+ * @return A pair of row bitmask and count of valid rows.
+ */
+std::pair<rmm::device_buffer, cudf::size_type> get_valid_row_info(cudf::table_view const& build,
```

Comment:
Minor suggestion: the code might be more readable if you made this a small helper struct with two visible attributes `composite_bitmask` and `valid_num_rows` rather than a function returning a pair.